libs/wingsdk/src/target-sim/redis.inflight.ts

```diff
@@ -0,0 +1,131 @@
+import Dockerode from "dockerode";
+import { v4 as uuidv4 } from "uuid";
+import { ISimulatorResourceInstance } from "./resource";
+import { RedisAttributes, RedisSchema } from "./schema-resources";
+import { IRedisClient } from "../redis";
+import { ISimulatorContext } from "../testing/simulator";
+
+// Issue using types from ioredis with JSII
+//import { Redis as IoRedis } from "ioredis";
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const IoRedis = require("ioredis");
+
+export class Redis implements IRedisClient, ISimulatorResourceInstance {
+  private container_name: string;
+  private readonly context: ISimulatorContext;
+
+  private connection_url?: string = undefined;
+  private docker?: Dockerode = undefined;
+  private connection?: any;
+
+  public constructor(_props: RedisSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+    this.container_name = `wing-sim-redis-${this.context.resourcePath.replace(
+      "/",
+      "."
+    )}-${uuidv4()}`;
+  }
+
+  public async init(): Promise<RedisAttributes> {
+    this.docker = new Dockerode();
+    // Create a redis container
+    try {
+      const container = await this.docker!.createContainer({
+        Image: "redis",
+        name: this.container_name,
+        HostConfig: {
+          PortBindings: {
+            "6379/tcp": [
+              {
+                HostPort: "0", // let docker pick a port
+              },
+            ],
+          },
+        },
+      });
+
+      // Start the redis container
+      await container.start();
+      // Inspect the container to get the host port
+      const hostPort = (await container.inspect()).NetworkSettings.Ports[
+        "6379/tcp"
+      ][0].HostPort;
+
+      // redis url based on host port
+      this.connection_url = `redis://0.0.0.0:${hostPort!}`;
+      return {};
+    } catch (e) {
+      throw Error(`Error setting up Redis resource simulation (${e})
+      - Make sure you have docker installed and running
+      - Make sure you have the redis image installed ('docker pull redis')`);
```

Comment:
Why not pull the image as part of execution of the resource?
Since we have access to the docker API, I don't think it's super useful to ask the user to do it.

Additionally, the image should be pinned to a very specific hash. This improves cacheability and reduces possible supply chain issues. I don't think people need the flexibility of choosing their own redis image in the sim.